sklearn/datasets/value_dropper.py

```diff
@@ -0,0 +1,300 @@
+# Author : Raghav RV <rvraghav93@gmail.com>
+#
+# Licence : BSD 3 clause
+
+import numpy as np
+import numbers
+
+from sklearn.utils import check_array
+from sklearn.utils import check_random_state
+from sklearn.utils.multiclass import type_of_target
+
+from sklearn.base import TransformerMixin
+from sklearn.preprocessing import LabelEncoder
+
+
+__all__ = ["ValueDropper"]
+
+
+class ValueDropper(TransformerMixin):
```

Comment:
I've for a while been confused about the availability of function alternatives in scikit-learn. We are a bit inconsistent. In a lot of places, functionality is duplicated from estimators to functions. We can't perform much common testing for functions. But in this case, we're applying something independent of the train-test split, and it's a new kind of thing, associated with the dataset module where there aren't any estimators. I don't see why you'd want to perform a grid search over varied `ValueDropper` params, either. A function seems a more natural form. But I'm ambivalent!